scripts/protocgen.sh

```diff
@@ -17,14 +17,29 @@ Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:. \
   -I "third_party/proto" \
   --grpc-gateway_out=logtostderr=true:. \
   $(find "${dir}" -maxdepth 1 -name '*.proto')
+
+  # generate swagger only for query files
+  query_file=$(find "${dir}" -maxdepth 1 -name 'query.proto')
+  if [[ ! -z "$query_file" ]]; then
+    protoc  \
+    -I "proto" \
+    -I "third_party/proto" \
+    "$query_file" \
+    --swagger_out=logtostderr=true,stderrthreshold=1000,fqn_for_swagger_name=true,simple_operation_ids=true:.
+  fi
 done
 
+# combine swagger files
+# uses nodejs package `swagger-combine`.
+swagger-combine ./client/grpc-gateway/config.json -o ./client/grpc-gateway/swagger.json --continueOnConflictingPaths true --includeDefinitions true
```

Comment:
A separate target should be OK. It sucks we need nodeJS to do this though. Is it hard to do manually?

Reply:
Sure, will move to a separate target. I tried to use grpc gateway's default merge option and `go-swagger`. Both are failing due to `Conflicts` with names (`Coin`, `Params` etc are used across all the modules and that is creating some issue). We can try a custom written script to do this may be.

Reply:
Yeah this is the exact same issue I ran into when trying my hand at automated swagger a long time ago before Proto even started. Sucks to see the issue still exist.

Anyway, if it's not terribly difficult, a script would be nice and I'd like to see that before merging. Not that I have any issues with NodeJS, but it just seems like overkill for this task IMHO.

Reply:
@alexanderbez @amaurymartiny we might need more time to fix the script, it's still in progress. If its not a blocker, I suggest to merge this PR and create a tracking issue for the custom golang script. wdyt?

Reply:
+1, if it's too involved, let's use the nodejs package right now and think about a script later on.